app/uk/gov/hmrc/economiccrimelevyregistration/navigation/CtUtrPageNavigator.scala

```diff
@@ -18,22 +18,43 @@ package uk.gov.hmrc.economiccrimelevyregistration.navigation
 import play.api.mvc.Call
 import uk.gov.hmrc.economiccrimelevyregistration.models.{CheckMode, NormalMode, Registration}
 import uk.gov.hmrc.economiccrimelevyregistration.controllers.routes
+import uk.gov.hmrc.economiccrimelevyregistration.models.OtherEntityType.Trust
 
 class CtUtrPageNavigator extends PageNavigator {
   override protected def navigateInNormalMode(registration: Registration): Call =
-    routes.CtUtrPostcodeController.onPageLoad(NormalMode)
+    registration.otherEntityJourneyData.ctUtr match {
+      case None    => routes.NotableErrorController.answersAreInvalid()
+      case Some(_) =>
+        registration.otherEntityJourneyData.entityType match {
+          case None        => routes.NotableErrorController.answersAreInvalid()
+          case Some(value) =>
```

Comment:
I _think_ you can consolidate this logic into the case condition using something like case Some(value @ Trust) => and then Some(value) => will handle all other values that are not a Trust.

Reply:
Fixed this. Thank you for the suggestion, this is the first time that I've had a chance to learn about variable binding in pattern matching. I was always putting that away from some other time 😄